### PR TITLE
feat(repo): Add initial indexer project with SurrealDB

### PR DESCRIPTION
**Pull request by pedronauck**

### CODE REVIEW

File: crates/fuel-indexer/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+            filter_subjects: vec![Transaction::WILDCARD_LIST[0].to_string()],
+        })
+        .await?;
+
+    while let Some(msg) = subscription.next().await {
+        let msg = msg?;
+        let transaction = Transaction::decode(msg.payload.clone().into()).await;
+        let tx_id = transaction.id(&ChainId::default());
+        let id = format!("0x{}", tx_id);
+        let thing_id = ("transaction".to_string(), id.clone());
```
Comment (0xterminator) [inline crates/fuel-indexer/src/main.rs:111]: I dont quite understand the variable naming here - thing_id ? (same above)

Comment (0xterminator) [reply crates/fuel-indexer/src/main.rs:111]: Maybe key ?

File: crates/fuel-indexer/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    let db_url =
+        dotenvy::var("SURREALDB_URL").expect("`SURREALDB_URL` env must be set");
+    let db_user = dotenvy::var("SURREALDB_USER")
+        .expect("`SURREALDB_USER` env must be set");
+    let db_pass = dotenvy::var("SURREALDB_PASS")
+        .expect("`SURREALDB_PASS` env must be set");
+
+    let db = Surreal::new::<Ws>(db_url).await?;
```
Comment (0xterminator) [inline crates/fuel-indexer/src/main.rs:40]: Does Ws stand for websockets ?

**Review by Jurshsmith — APPROVED** [review]

**Comment by 0xterminator** [comment]
@pedronauck Once you rebase and fix the conflicts, I will approve ant we can merge it

**Pull request merged**